### ISSUE TITLE
Monitoring data import not working to ESS

### ISSUE THREAD

**Issue by bosinm** [elastic/support-diagnostics#450]
This is a follow up of this [issue](https://github.com/elastic/support-diagnostics/issues/350) (attempting to push monitoring data to an ESS cluster with monitoring enabled).

- Support diags version 8.1.2
- ESS 7.7.1 with monitoring enabled

```
support-diagnostics-8.1.2 miguelb$ sudo ./import-monitoring.sh --host 92dec3103603444cbdcf98e8ec185309.europe-west3.gcp.cloud.es.io --port 9243 -u elastic -p --ssl -i /Users/miguelb/Desktop/tickets/stuffreviewed/TESTINGbuilds/support-diagnostics-8.1.2/monitoring-export-20201027-013504.zip
Password:
Using /usr/bin/java as Java Runtime
Using -Xms256m -Xmx2000m  for options.
2020-10-27 01:56:49,696 main ERROR Unable to locate appender "diag" for logger config "root"
Processing diagnosticInputs...
Elasticsearch user password: ************************
Deleted directory: /Users/miguelb/Desktop/tickets/stuffreviewed/TESTINGbuilds/support-diagnostics-8.1.2/monitoring-export.
Creating temporary directory /Users/miguelb/Desktop/tickets/stuffreviewed/TESTINGbuilds/support-diagnostics-8.1.2/monitoring-export
Configuring log file.
Diagnostic logger reconfigured for inclusion into archive
Extract completed successfully!
Processing: indices_stats.json
Indexing document batch 0 to 291
Error processing JSON event for {}.
java.lang.RuntimeException: Broken pipe
	at com.elastic.support.rest.RestClient.execRequest(RestClient.java:79) ~[support-diagnostics-8.1.2.jar:8.1.2]
	at com.elastic.support.rest.RestClient.execPost(RestClient.java:91) ~[support-diagnostics-8.1.2.jar:8.1.2]
	at com.elastic.support.monitoring.MonitoringImportProcessor.writeBatch(MonitoringImportProcessor.java:145) ~[support-diagnostics-8.1.2.jar:8.1.2]
	at com.elastic.support.monitoring.MonitoringImportProcessor.process(MonitoringImportProcessor.java:124) [support-diagnostics-8.1.2.jar:8.1.2]
	at com.elastic.support.monitoring.MonitoringImportProcessor.exec(MonitoringImportProcessor.java:54) [support-diagnostics-8.1.2.jar:8.1.2]
	at com.elastic.support.monitoring.MonitoringImportService.execImport(MonitoringImportService.java:55) [support-diagnostics-8.1.2.jar:8.1.2]
	at com.elastic.support.monitoring.MonitoringImportApp.main(MonitoringImportApp.java:37) [support-diagnostics-8.1.2.jar:8.1.2]
0 events written from indices_stats.json
Processing: index_stats.json
Indexing document batch 0 to 500
Error processing JSON event for {}.
java.lang.RuntimeException: Broken pipe
	at com.elastic.support.rest.RestClient.execRequest(RestClient.java:79) ~[support-diagnostics-8.1.2.jar:8.1.2]
	at com.elastic.support.rest.RestClient.execPost(RestClient.java:91) ~[support-diagnostics-8.1.2.jar:8.1.2]
	at com.elastic.support.monitoring.MonitoringImportProcessor.writeBatch(MonitoringImportProcessor.java:145) 
```
From the extract log(attached):

```
01:57:01.611 [main] INFO  com.elastic.support.BaseService - Diagnostic logger reconfigured for inclusion into archive
01:57:03.718 [main] INFO  org.apache.http.impl.execchain.RetryExec - I/O exception (java.net.SocketException) caught when processing request to {s}->https://92dec3103603444cbdcf98e8ec185309.europe-west3.gcp.cloud.es.io:9243: Broken pipe
01:57:03.718 [main] INFO  org.apache.http.impl.execchain.RetryExec - Retrying request to {s}->https://92dec3103603444cbdcf98e8ec185309.europe-west3.gcp.cloud.es.io:9243
01:57:03.949 [main] INFO  org.apache.http.impl.execchain.RetryExec - I/O exception (java.net.SocketException) caught when processing request to {s}->https://92dec3103603444cbdcf98e8ec185309.europe-west3.gcp.cloud.es.io:9243: Broken pipe
01:57:03.949 [main] INFO  org.apache.http.impl.execchain.RetryExec - Retrying request to {s}->https://92dec3103603444cbdcf98e8ec185309.europe-west3.gcp.cloud.es.io:9243
01:57:04.149 [main] INFO  org.apache.http.impl.execchain.RetryExec - I/O exception (java.net.SocketException) caught when processing request to {s}->https://92dec3103603444cbdcf98e8ec185309.europe-west3.gcp.cloud.es.io:9243: Broken pipe
01:57:04.149 [main] INFO  org.apache.http.impl.execchain.RetryExec - Retrying request to {s}->https://92dec3103603444cbdcf98e8ec185309.europe-west3.gcp.cloud.es.io:9243
01:57:04.341 [main] ERROR com.elastic.support.rest.RestClient - Unexpected Execution Error
java.net.SocketException: Broken pipe
	at java.net.SocketOutputStream.socketWrite0(Native Method) ~[?:1.8.0_60]
	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:109) ~[?:1.8.0_60]
	at java.net.SocketOutputStream.write(SocketOutputStream.java:153) ~[?:1.8.0_60]
	at sun.security.ssl.OutputRecord.writeBuffer(OutputRecord.java:431) ~[?:1.8.0_60]
	at sun.security.ssl.OutputRecord.write(OutputRecord.java:417) ~[?:1.8.0_60]
	at sun.security.ssl.SSLSocketImpl.writeRecordInternal(SSLSocketImpl.java:876) ~[?:1.8.0_60]
	at sun.security.ssl.SSLSocketImpl.writeRecord(SSLSocketImpl.java:847) ~[?:1.8.0_60]
	at sun.security.ssl.AppOutputStream.write(AppOutputStream.java:123) ~[?:1.8.0_60]
```
[import.log](https://github.com/elastic/support-diagnostics/files/5442337/import.log)

Are we missing something here?

**Comment by CBlaisure** [comment]
Are you able to recreate this?  After review, `java.lang.RuntimeException: Broken pipe` implies a networking issue and therefore might be temporary

**Comment by nellicus** [comment]
@bosinm assuming this is not reproducible and we'll close. please reopen if you can repro thanks.

**Comment by toby-sutor** [comment]
Reopening as I can still reproduce this.

**Steps to reproduce**:
1. Create an ESS deployment (7.10.1 on GCP Belgium, highIO with a single 4GB node).
2. Enable Stack Monitoring (same behavior with skipping this step)
2. Download the Support Diagnostics from https://github.com/elastic/support-diagnostics/releases/tag/8.1.2
3. Run it like this (on a company Mac with Firewall enabled/disabled):
`./import-monitoring.sh -i ../monitoring-export-20201224-125359.zip --host bd1d5851e5d24b1b98cf529f0a888439.europe-west1.gcp.cloud.es.io -u elastic -p --ssl --port 9243` (the sample can be found [https://elastic.my.salesforce.com/0694M00000A5kBP](here))
Returns:
````
$ ./import-monitoring.sh -i ../monitoring-export-20201224-125359.zip --host bd1d5851e5d24b1b98cf529f0a888439.europe-west1.gcp.cloud.es.io -u elastic -p --ssl --port 9243
Using /usr/bin/java as Java Runtime
Using -Xms256m -Xmx2000m  for options.
2020-12-28 11:19:24,179 main ERROR Unable to locate appender "diag" for logger config "root"
Processing diagnosticInputs...
Elasticsearch user password: ************************
Deleted directory: /Users/toby/ID4H5DQbSPoanKZH6x06PueEsupport-diagnostics-8.1.2/monitoring-export.
Creating temporary directory /Users/toby/support-diagnostics-8.1.2/monitoring-export
Configuring log file.
Diagnostic logger reconfigured for inclusion into archive
Extract completed successfully!
Processing: metricbeat-cpu.json
Indexing document batch 0 to 500
Error processing JSON event for {}.
java.lang.RuntimeException: Broken pipe (Write failed)
	at com.elastic.support.rest.RestClient.execRequest(RestClient.java:79) ~[support-diagnostics-8.1.2.jar:8.1.2]
	at com.elastic.support.rest.RestClient.execPost(RestClient.java:91) ~[support-diagnostics-8.1.2.jar:8.1.2]
	at com.elastic.support.monitoring.MonitoringImportProcessor.writeBatch(MonitoringImportProcessor.java:145) ~[support-diagnostics-8.1.2.jar:8.1.2]
	at com.elastic.support.monitoring.MonitoringImportProcessor.process(MonitoringImportProcessor.java:112) [support-diagnostics-8.1.2.jar:8.1.2]
	at com.elastic.support.monitoring.MonitoringImportProcessor.exec(MonitoringImportProcessor.java:54) [support-diagnostics-8.1.2.jar:8.1.2]
	at com.elastic.support.monitoring.MonitoringImportService.execImport(MonitoringImportService.java:55) [support-diagnostics-8.1.2.jar:8.1.2]
	at com.elastic.support.monitoring.MonitoringImportApp.main(MonitoringImportApp.java:37) [support-diagnostics-8.1.2.jar:8.1.2]
0 events written from metricbeat-cpu.json
Processing: metricbeat-diskio.json
Indexing document batch 0 to 500
Error processing JSON event for {}.
java.lang.RuntimeException: Broken pipe (Write failed)
````

From the logs:
````
11:29:47.618 [main] INFO  com.elastic.support.BaseService - Diagnostic logger reconfigured for inclusion into archive
11:30:42.406 [main] ERROR com.elastic.support.rest.RestClient - Unexpected Execution Error
javax.net.ssl.SSLException: Broken pipe (Write failed)
	at sun.security.ssl.Alert.createSSLException(Alert.java:127) ~[?:?]
	at sun.security.ssl.TransportContext.fatal(TransportContext.java:320) ~[?:?]
	at sun.security.ssl.TransportContext.fatal(TransportContext.java:263) ~[?:?]
	at sun.security.ssl.TransportContext.fatal(TransportContext.java:258) ~[?:?]
	at sun.security.ssl.SSLSocketImpl$AppOutputStream.write(SSLSocketImpl.java:1016) ~[?:?]
	at org.apache.http.impl.io.SessionOutputBufferImpl.streamWrite(SessionOutputBufferImpl.java:124) ~[httpcore-4.4.12.jar:4.4.12]
	at org.apache.http.impl.io.SessionOutputBufferImpl.write(SessionOutputBufferImpl.java:160) ~[httpcore-4.4.12.jar:4.4.12]
	at org.apache.http.impl.io.ContentLengthOutputStream.write(ContentLengthOutputStream.java:113) ~[httpcore-4.4.12.jar:4.4.12]
	at org.apache.http.impl.io.ContentLengthOutputStream.write(ContentLengthOutputStream.java:120) ~[httpcore-4.4.12.jar:4.4.12]
	at org.apache.http.entity.StringEntity.writeTo(StringEntity.java:167) ~[httpcore-4.4.12.jar:4.4.12]
	at org.apache.http.impl.DefaultBHttpClientConnection.sendRequestEntity(DefaultBHttpClientConnection.java:156) ~[httpcore-4.4.12.jar:4.4.12]
	at org.apache.http.impl.conn.CPoolProxy.sendRequestEntity(CPoolProxy.java:152) ~[httpclient-4.5.10.jar:4.5.10]
	at org.apache.http.protocol.HttpRequestExecutor.doSendRequest(HttpRequestExecutor.java:238) ~[httpcore-4.4.12.jar:4.4.12]
	at org.apache.http.protocol.HttpRequestExecutor.execute(HttpRequestExecutor.java:123) ~[httpcore-4.4.12.jar:4.4.12]
	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:272) ~[httpclient-4.5.10.jar:4.5.10]
	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186) ~[httpclient-4.5.10.jar:4.5.10]
	at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89) ~[httpclient-4.5.10.jar:4.5.10]
	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110) ~[httpclient-4.5.10.jar:4.5.10]
	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185) ~[httpclient-4.5.10.jar:4.5.10]
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:72) ~[httpclient-4.5.10.jar:4.5.10]
	at com.elastic.support.rest.RestClient.execRequest(RestClient.java:73) [support-diagnostics-8.1.2.jar:8.1.2]
	at com.elastic.support.rest.RestClient.execPost(RestClient.java:91) [support-diagnostics-8.1.2.jar:8.1.2]
	at com.elastic.support.monitoring.MonitoringImportProcessor.checkForExtractTemplates(MonitoringImportProcessor.java:166) [support-diagnostics-8.1.2.jar:8.1.2]
	at com.elastic.support.monitoring.MonitoringImportProcessor.<init>(MonitoringImportProcessor.java:43) [support-diagnostics-8.1.2.jar:8.1.2]
	at com.elastic.support.monitoring.MonitoringImportService.execImport(MonitoringImportService.java:54) [support-diagnostics-8.1.2.jar:8.1.2]
	at com.elastic.support.monitoring.MonitoringImportApp.main(MonitoringImportApp.java:37) [support-diagnostics-8.1.2.jar:8.1.2]
	Suppressed: java.net.SocketException: Broken pipe (Write failed)
		at java.net.SocketOutputStream.socketWrite0(Native Method) ~[?:?]
		at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:110) ~[?:?]
		at java.net.SocketOutputStream.write(SocketOutputStream.java:150) ~[?:?]
		at sun.security.ssl.SSLSocketOutputRecord.encodeAlert(SSLSocketOutputRecord.java:81) ~[?:?]
		at sun.security.ssl.TransportContext.fatal(TransportContext.java:351) ~[?:?]
		at sun.security.ssl.TransportContext.fatal(TransportContext.java:263) ~[?:?]
		at sun.security.ssl.TransportContext.fatal(TransportContext.java:258) ~[?:?]
		at sun.security.ssl.SSLSocketImpl$AppOutputStream.write(SSLSocketImpl.java:1016) ~[?:?]
		at org.apache.http.impl.io.SessionOutputBufferImpl.streamWrite(SessionOutputBufferImpl.java:124) ~[httpcore-4.4.12.jar:4.4.12]
		at org.apache.http.impl.io.SessionOutputBufferImpl.write(SessionOutputBufferImpl.java:160) ~[httpcore-4.4.12.jar:4.4.12]
		at org.apache.http.impl.io.ContentLengthOutputStream.write(ContentLengthOutputStream.java:113) ~[httpcore-4.4.12.jar:4.4.12]
		at org.apache.http.impl.io.ContentLengthOutputStream.write(ContentLengthOutputStream.java:120) ~[httpcore-4.4.12.jar:4.4.12]
		at org.apache.http.entity.StringEntity.writeTo(StringEntity.java:167) ~[httpcore-4.4.12.jar:4.4.12]
		at org.apache.http.impl.DefaultBHttpClientConnection.sendRequestEntity(DefaultBHttpClientConnection.java:156) ~[httpcore-4.4.12.jar:4.4.12]
		at org.apache.http.impl.conn.CPoolProxy.sendRequestEntity(CPoolProxy.java:152) ~[httpclient-4.5.10.jar:4.5.10]
		at org.apache.http.protocol.HttpRequestExecutor.doSendRequest(HttpRequestExecutor.java:238) ~[httpcore-4.4.12.jar:4.4.12]
		at org.apache.http.protocol.HttpRequestExecutor.execute(HttpRequestExecutor.java:123) ~[httpcore-4.4.12.jar:4.4.12]
		at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:272) ~[httpclient-4.5.10.jar:4.5.10]
		at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186) ~[httpclient-4.5.10.jar:4.5.10]
		at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89) ~[httpclient-4.5.10.jar:4.5.10]
		at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110) ~[httpclient-4.5.10.jar:4.5.10]
		at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185) ~[httpclient-4.5.10.jar:4.5.10]
		at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:72) ~[httpclient-4.5.10.jar:4.5.10]
		at com.elastic.support.rest.RestClient.execRequest(RestClient.java:73) [support-diagnostics-8.1.2.jar:8.1.2]
		at com.elastic.support.rest.RestClient.execPost(RestClient.java:91) [support-diagnostics-8.1.2.jar:8.1.2]
		at com.elastic.support.monitoring.MonitoringImportProcessor.checkForExtractTemplates(MonitoringImportProcessor.java:166) [support-diagnostics-8.1.2.jar:8.1.2]
		at com.elastic.support.monitoring.MonitoringImportProcessor.<init>(MonitoringImportProcessor.java:43) [support-diagnostics-8.1.2.jar:8.1.2]
		at com.elastic.support.monitoring.MonitoringImportService.execImport(MonitoringImportService.java:54) [support-diagnostics-8.1.2.jar:8.1.2]
		at com.elastic.support.monitoring.MonitoringImportApp.main(MonitoringImportApp.java:37) [support-diagnostics-8.1.2.jar:8.1.2]
Caused by: java.net.SocketException: Broken pipe (Write failed)
	at java.net.SocketOutputStream.socketWrite0(Native Method) ~[?:?]
	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:110) ~[?:?]
	at java.net.SocketOutputStream.write(SocketOutputStream.java:150) ~[?:?]
	at sun.security.ssl.SSLSocketOutputRecord.deliver(SSLSocketOutputRecord.java:320) ~[?:?]
	at sun.security.ssl.SSLSocketImpl$AppOutputStream.write(SSLSocketImpl.java:1011) ~[?:?]
	... 21 more
11:30:42.411 [main] ERROR com.elastic.support.monitoring.MonitoringImportProcessor - Issue checking template metricbeat-system-diag
java.lang.RuntimeException: Broken pipe (Write failed)
````

**Comment by eedugon** [comment]
This was a real issue (and easy to reproduce), but we have closed it (and similar reports) and fixed in 9.0.0 without details...

Anyway this looks fixed in 9.0.0 pre-release, check this nice comment for more info and details on how to execute the import:
https://github.com/elastic/support-diagnostics/issues/457#issuecomment-780844011
